homeassistant/components/melnor/time.py

```diff
@@ -0,0 +1,92 @@
+"""Number support for Melnor Bluetooth water timer."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from datetime import time
+from typing import Any
+
+from melnor_bluetooth.device import Valve
+
+from homeassistant.components.time import TimeEntity, TimeEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .models import (
+    MelnorDataUpdateCoordinator,
+    MelnorZoneEntity,
+    get_entities_for_valves,
+)
+
+
+@dataclass
+class MelnorZoneTimeEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    set_time_fn: Callable[[Valve, time], Coroutine[Any, Any, None]]
+    state_fn: Callable[[Valve], Any]
+
+
+@dataclass
+class MelnorZoneTimeEntityDescription(
+    TimeEntityDescription, MelnorZoneTimeEntityDescriptionMixin
+):
+    """Describes Melnor number entity."""
+
+
+ZONE_ENTITY_DESCRIPTIONS: list[MelnorZoneTimeEntityDescription] = [
+    MelnorZoneTimeEntityDescription(
+        entity_category=EntityCategory.CONFIG,
+        icon="mdi:calendar-clock-outline",
```

Comment:
The default icon for time integration should already be a clock. If that is not the case right now, that's a bug and should be fixed in frontend.
```suggestion
```